cloud/pkg/dynamiccontroller/application/application.go

```diff
@@ -493,7 +509,13 @@ func (c *Center) ProcessApplication(app *Application) (interface{}, error) {
 		if err := app.ReqBodyTo(obj); err != nil {
 			return nil, err
 		}
-		retObj, err := c.kubeclient.Resource(gvr).Namespace(ns).Update(context.TODO(), obj, *option)
+		var retObj interface{}
+		var err error
+		if app.Subresource == "" {
+			retObj, err = c.kubeclient.Resource(gvr).Namespace(ns).Update(context.TODO(), obj, *option)
+		} else {
+			retObj, err = c.kubeclient.Resource(gvr).Namespace(ns).Update(context.TODO(), obj, *option, app.Subresource)
+		}
```

Comment:
Looks good. In K8s, for create method, if it has no `subresource`, the path can't contain {name} parameter.
And for the update method, it must contain {name} parameter. 

Ref: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L470